internal/aks/delete.go

```diff
@@ -0,0 +1,48 @@
+package aks
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-11-01/containerservice"
+	"github.com/Azure/go-autorest/autorest/to"
+	aksv1 "github.com/rancher/aks-operator/pkg/apis/aks.cattle.io/v1"
+	"github.com/sirupsen/logrus"
+)
+
+// Delete AKS managed Kubernetes cluster
+func RemoveCluster(ctx context.Context, clusterClient *containerservice.ManagedClustersClient, spec *aksv1.AKSClusterConfigSpec) (err error) {
+	future, err := clusterClient.Delete(ctx, spec.ResourceGroup, spec.ClusterName)
+	if err != nil {
+		return err
+	}
+
+	err = future.WaitForCompletionRef(ctx, clusterClient.Client)
```

Comment:
Something to think about for a different PR, but I was having trouble deleting clusters from Rancher because the `cluster` object gets deleted immediately but the `aksclusterconfig` object blocks while being deleted here. Maybe the controller could set a "deleting" state that the cluster handler can check before deleting?

Reply:
Yes, let's do this in another PR. For tracking I have added it to https://github.com/rancher/rancher/issues/31648#issuecomment-829946108